pkg/restapi/rp/operation/operations.go

```diff
@@ -586,3 +592,61 @@ func testResponse(w io.Writer) {
 		fmt.Printf("error writing test response: %s", err.Error())
 	}
 }
+
+func (o *Operation) createRPTenant(w http.ResponseWriter, r *http.Request) {
+	request := &CreateRPTenantRequest{}
+
+	err := json.NewDecoder(r.Body).Decode(request)
+	if err != nil {
+		msg := fmt.Sprintf("failed to decode request: %s", err)
+		logger.Errorf(msg)
+		commhttp.WriteErrorResponse(w, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	if request.ClientID == "" || request.Label == "" || request.PublicDID == "" {
```

Comment:
Here publicDID is part of the request.  Who calls this API ? Who creates public DID ?

Reply:
This should be a protected endpoint. It should be called by someone with the necessary privileges to register new RP tenants.

As for the publicDID - we can have the adapter create it while fulfilling this request.

Reply:
@rolsonquadras create #67